packages/eds-core-react/src/components/Datepicker/calendars/YearGrid.tsx

```diff
@@ -0,0 +1,109 @@
+// Disable no-autofocus - it's not the native autofocus attribute, but react-aria's autoFocus prop
+/* eslint-disable jsx-a11y/no-autofocus */
+import styled from 'styled-components'
+import { tokens } from '@equinor/eds-tokens'
+import { FocusScope, useFocusManager } from 'react-aria'
+import { KeyboardEvent } from 'react'
+
+const Grid = styled.div`
+  display: grid;
+  grid-template-columns: repeat(6, 1fr);
+  grid-gap: 8px;
+  margin: 8px;
+`
+
+const GridColumn = styled.button<{ $active: boolean }>`
+  background-color: transparent;
+  outline: none;
+  border: none;
+  display: flex;
+  justify-content: center;
+  cursor: pointer;
+  padding: 8px;
+  font-size: ${tokens.typography.navigation.button.fontSize};
+  font-family: ${tokens.typography.navigation.button.fontFamily};
+  font-weight: ${tokens.typography.navigation.button.fontWeight};
+  line-height: ${tokens.typography.navigation.button.lineHeight};
+  color: ${tokens.colors.text.static_icons__default.rgba};
+  border-radius: 999px;
+  ${({ $active }) =>
+    $active
+      ? `background-color: ${tokens.colors.interactive.primary__selected_highlight.rgba}`
+      : ''};
+  &:hover {
+    background-color: #f0f0f0;
+  }
+  &:focus {
+    outline: 2px dashed ${tokens.colors.interactive.primary__resting.rgba};
+  }
+`
+
+const GridFocusManager = ({
+  year: selectedYear,
+  setFocusedYear,
+}: {
+  setFocusedYear: (year: number) => void
+  year: number
+}) => {
+  const focusManager = useFocusManager()
+  const onKeyDown = (e: KeyboardEvent<HTMLButtonElement>) => {
+    const target = e.currentTarget
+    const parent = target.parentElement as HTMLDivElement
+    switch (e.key) {
+      case 'ArrowRight':
+        e.preventDefault()
```

Comment:
Can this be put at the top shared?

Reply:
Sorry, not sure I understand the comment?

Reply:
Because we only want to call prevetnDefault on the keys we handle ourselves (I think?)